Remote copy

ansarars opened the pull request.

farhan7500 commented on Modules/hpe3par_remote_copy.py:
```diff
@@ -389,12 +389,78 @@ def recover_remote_copy_group(
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        response = client_obj.admitRemoteCopyLinks(target_name, source_port, target_port_wwn_or_ip)
```
What happens if a remote copy link is already present? Is there a way to check that before we attempt to add the link?

farhan7500 commented on Modules/hpe3par_remote_copy.py:
```diff
@@ -389,12 +389,78 @@ def recover_remote_copy_group(
+    try:
+        client_obj.login(storage_system_username, storage_system_password)
+        client_obj.setSSHOptions(storage_system_ip, storage_system_username, storage_system_password)
+        response = client_obj.dismissRemoteCopyLinks(target_name, source_port, target_port_wwn_or_ip)
```
Similar to above, what happens if the link is not admitted at all and we try to remove it?

farhan7500 commented:
Once your're done, please attached the test logs here

ansarars commented:
Case 1:- Admit rcopy link
===============
CSSOS-SSA04 cli% showrcopy
===========================
Remote Copy System Information
Status: Started, Normal

Target Information

Name        ID Type Status Options Policy
CSSOS-SSA05  2 IP   ready  -       mirror_config
Romulus      3 IP   new    -       mirror_config

Link Information

Target      Node  Address     Status Options
CSSOS-SSA05 1:3:1 192.168.2.2 Up     -
receive     1:3:1 receive     Up     -



Ansible playbook execution for admitlink:-
=========================================
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module# ansible-playbook admit_link.yml
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

 [WARNING]: While constructing a mapping from /home/Arshad/workspace/hpe3par_ansible_module/admit_link.yml, line 3, column 5, found a duplicate dict key (http_proxy). Using
last defined value only.


PLAY [localhost] ****************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************
ok: [localhost]

TASK [Admit remote copy link] ***************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0

root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#


showrcopy output after admitlink:-
=================================
CSSOS-SSA04 cli% showrcopy

Remote Copy System Information
Status: Started, Normal

Target Information

Name        ID Type Status Options Policy
CSSOS-SSA05  2 IP   ready  -       mirror_config
Romulus      3 IP   new    -       mirror_config

Link Information

Target      Node  Address     Status Options
CSSOS-SSA05 0:3:1 192.168.1.2 Up     -
CSSOS-SSA05 1:3:1 192.168.2.2 Up     -
receive     0:3:1 receive     Up     -
receive     1:3:1 receive     Up     -


Case 2:-
============
Ansible playbook execution for dismisslink:-
==============================================
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module# ansible-playbook dismiss_link.yml
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

 [WARNING]: While constructing a mapping from /home/Arshad/workspace/hpe3par_ansible_module/dismiss_link.yml, line 3, column 5, found a duplicate dict key (http_proxy). Using
last defined value only.


PLAY [localhost] ****************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************
ok: [localhost]

TASK [dismiss remote copy link] *************************************************************************************************************************************************
changed: [localhost]

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0

root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#


showrcopy after dismiss link:-
===============================
CSSOS-SSA04 cli% showrcopy

Remote Copy System Information
Status: Started, Normal

Target Information

Name        ID Type Status Options Policy
CSSOS-SSA05  2 IP   ready  -       mirror_config
Romulus      3 IP   new    -       mirror_config

Link Information

Target      Node  Address     Status Options
CSSOS-SSA05 1:3:1 192.168.2.2 Up     -
receive     1:3:1 receive     Up     -


Case 3:-
=================
When link is not present then executing dismiss link:-
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module# ansible-playbook dismiss_link.yml
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

 [WARNING]: While constructing a mapping from /home/Arshad/workspace/hpe3par_ansible_module/dismiss_link.yml, line 3, column 5, found a duplicate dict key (http_proxy). Using
last defined value only.


PLAY [localhost] ****************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************
ok: [localhost]

TASK [dismiss remote copy link] *************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0

root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#


Case 4:-
==========
When link is already present and executed admit link:-
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module# ansible-playbook admit_link.yml
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

 [WARNING]: While constructing a mapping from /home/Arshad/workspace/hpe3par_ansible_module/admit_link.yml, line 3, column 5, found a duplicate dict key (http_proxy). Using
last defined value only.


PLAY [localhost] ****************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************
ok: [localhost]

TASK [Admit remote copy link] ***************************************************************************************************************************************************
ok: [localhost]

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0

root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#
root@CSSOSBE03-B05:/home/Arshad/workspace/hpe3par_ansible_module#
